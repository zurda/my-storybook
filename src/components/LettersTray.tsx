import React from 'react'
import styled from '@emotion/styled'
import { colours } from '../style/shared'

const LettersContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: 16px auto;
`

export const Char = styled.button`
  padding: 8px;
  font-weight: bold;
  margin: 2px;
  width: 10%;
  background-color: ${(props) => (props.disabled ? 'white' : colours.pictonBlue)};
  color: ${(props) => (props.disabled ? colours.diamond : 'white')};
  box-shadow: 0px 3px 15px rgba(0, 0, 0, 0.2);
  border-radius: 2px;
  border: 0;
  @media (min-width: 420px) {
    font-size: 18px;
  }
`

export const LettersTray = ({ guessedLetters, onClickHandler }: any) => (
  <LettersContainer>
    {['A', 'B', 'C', 'D', 'E'].map((char, i) => (
      <Char
        onClick={() => onClickHandler(char)}
        key={i}
        value={char}
        disabled={guessedLetters.indexOf(char) > -1}
      >
        {char}
      </Char>
    ))}
  </LettersContainer>
)
