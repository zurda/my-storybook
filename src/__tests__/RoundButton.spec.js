import React from 'react'
import { create } from 'react-test-renderer'
import { RoundButton } from '../components/GameControllers'

describe('RoundButton component', () => {
  test('Matches the snapshot', () => {
    const roundButton = create(<RoundButton>Button text</RoundButton>)
    expect(roundButton.toJSON()).toMatchSnapshot()
  })
})
