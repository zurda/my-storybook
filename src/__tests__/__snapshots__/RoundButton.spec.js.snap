// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton component Matches the snapshot 1`] = `
<button
  className="css-a35l7o"
>
  Button text
</button>
`;
